pkg/canary/comparator/comparator.go

```diff
@@ -186,12 +244,116 @@ func (c *Comparator) run() {
 			c.entrySent(e)
 		case <-t.C:
 			c.pruneEntries()
+		case <-sc.C:
+			// Only run one instance of spot check at a time.
+			c.spotMtx.Lock()
+			if !c.spotCheckRunning {
+				c.spotCheckRunning = true
+				go c.spotCheckEntries(time.Now())
+			}
+			c.spotMtx.Unlock()
+		case <-mt.C:
+			// Only run one intstance of metric tests at a time.
+			c.metTestMtx.Lock()
+			if !c.metricTestRunning {
+				c.metricTestRunning = true
+				go c.metricTest(time.Now())
+			}
+			c.metTestMtx.Unlock()
 		case <-c.quit:
 			return
 		}
 	}
 }
 
+func (c *Comparator) metricTest(currTime time.Time) {
+	// Always make sure to set the running state back to false
+	defer func() {
+		c.metTestMtx.Lock()
+		c.metricTestRunning = false
+		c.metTestMtx.Unlock()
+	}()
+	adjustedRange := c.metricTestRange
+
+	// Adjust the query range to not be longer than the canary has been running.
+	// We can't query for 24 hours of counts if it's only been running for 10m,
+	// so we adjusted the range to the run time until we reach the desired lookback time.
+	if currTime.Add(-c.metricTestRange).Before(c.startTime) {
+		adjustedRange = currTime.Sub(c.startTime)
+	}
+	actualCount, err := c.rdr.QueryCountOverTime(fmt.Sprintf("%.0fs", adjustedRange.Seconds()))
+	if err != nil {
+		fmt.Fprintf(c.w, "error running metric query test: %s\n", err.Error())
+		return
+	}
+	expectedCount := float64(adjustedRange.Milliseconds()) / float64(c.writeInterval.Milliseconds())
+	deviation := expectedCount - actualCount
+	if deviation > 10 {
```

Comment:
why 10 ?

Reply:
why not?

Reply:
joking aside, i added this to debug some strange behavior i was seeing, there is some deviation to be expected in normal cases, say 3-4 values so 10 was a value I figured would only be hit for abnormal conditions